mysql-test/suite/s3/not_amazon.test

```diff
@@ -0,0 +1,35 @@
+--source include/have_s3.inc
+
+if (`SELECT @@s3_host_name LIKE "%.amazonaws.com"`)
```

Comment:
Not all Amazon S3 endpoints’ hostnames match `%.amazonaws.com`, in particular those in the AWS China regions. See https://docs.aws.amazon.com/general/latest/gr/s3.html for examples.

Reply:
I was not aware of that one. So. "%.amazonaws.com%" would match?

Reply:
That seems like it would _currently_ be sufficient, yes!